compiler/noirc_evaluator/src/ssa/ir/types.rs

```diff
@@ -297,6 +297,17 @@ impl Type {
             }
         }
     }
+
+    /// True if this is a function type or if it is a composite type which contains a function.
+    pub(crate) fn contains_function(&self) -> bool {
+        match self {
+            Type::Function => true,
+            Type::Numeric(_) | Type::Reference(_) => true,
```

Comment:
```suggestion
            Type::Numeric(_) | Type::Reference(_) => false,
```

Though I don't know if for references it should check if the referenced value is a function.

Reply:
> Though I don't know if for references it should check if the referenced value is a function.

It should - so it should recur on the element type. I think it should be possible to use references in globals... we haven't tested that at all though so it scares me a bit for mem2reg.

Reply:
doh 🤦 yes numeric types do not contain functions. We have to recur on the element type like Jake said so I updated this locally.